algorithms/schema-free/closeness_cent_approx_query.gsql

```diff
@@ -0,0 +1,238 @@
+CREATE QUERY closeness_approx (
+    SET<STRING> v_type, 
+    SET<STRING> e_type,
+		int k,  # sample num
+		int maxHops,  # max BFS explore steps 
+		double epsilon,  # error parameter
+    		bool print_accum, # output to console
+		string file_path,  # output file 
+		int debug,  # debug flag -- 0: No LOG;1: LOG without the sample-node bfs loop;2: ALL LOG.
+		int sample_index,  # random sample group
+		int maxsize,  # max size of connected components using exact closeness algorithm
+		bool wf = True # Wasserman and Faust formula 
+) {
+		
+		MinAccum<int> @cc_id;      # each vertex's tentative component id
+		MapAccum<int,int> @@cc_size;    # component size 
+		MapAccum<vertex,int> @@cc_map;   # closeness centrality of each node
+		SumAccum<int> @vid;  # internal id
+		SumAccum<int> @deltav;  # the distance from the pivot
+		SumAccum<int> @@curDist,@@dist;  # the distance of BFS algorithm 
+		MaxAccum<vertex> @pivot;  # the pivot of the node
+		SetAccum<vertex> @@sample;  # sample node set
+		OrAccum<bool> @sampled;  # sample node
+		OrAccum<bool> @visited,@visited0;  # visited node in BFS
+		SumAccum<int> @@totalDist;  # the sum distance of the sample node
+		SumAccum<int> @@nonsample;  # the size of non-sample node in the visiting node
+		MapAccum<vertex,int> @@cdist;  # the shortest distance between the sample node
+		AvgAccum @lavg; # LAVG: the average distance from the sample node within the threshold radius
+		SumAccum<int> @hcsum; # HCSUM: the sum distance from the sample node outside the threshold radius
+		SumAccum<int> @sdist; # record the distance when the sample node cannot be decided
+		SumAccum<int> @lnum; # number of nodes within the threshold radius
+		SumAccum<int> @hsum; # HSUM: the sum distance from non sample node outside the threshold radius
+		SumAccum<float> @totalDist; # the sum distance of the nodes
+		SumAccum<float> @closeness; # the closeness centrality
+		FILE f(file_path);
+		
+		int numVert,partition,i;
+		datetime t1,t2;
+	  float eps = 0.00001;
+	
+		Start = {v_type};
+		LOG(debug > 0,"query start",Start.size());
+		
+		# Total number of nodes
+    numVert = Start.size();
+	
+		# Initialize: Label each vertex with its own internal ID
+		comp = SELECT x FROM Start:x
+			POST-ACCUM x.@cc_id = getvid(x);
+
+		# Propagate smaller internal IDs until no more ID changes can be DOne
+		WHILE (comp.size()>0) DO
+			comp = SELECT t FROM comp:s-(e_type)->:t
+				ACCUM t.@cc_id += s.@cc_id 	// If s has a smaller id than t, copy the id to t
+	            HAVING t.@cc_id != t.@cc_id';
+		END;
+		LOG(debug > 0,"find connected components");
+	  
+		# get components size
+		Start = select s from Start:s
```

Comment:
Keywords should be capitalized. This is a code style that was established decades ago for SQL to improve readability, and it also helps some text editors which aren't programmed to recognize our custom GSQL language, but they can be set to color-code all the words in ALL CAPS.

Reply:
somehow reviewed the outdated commit, new commit has all keywords in caps